File: dongyimai-manager-web/src/main/webapp/js/controller/brandController.js
app.controller('brandController',function ($scope,$controller,brandService) {

    $controller('baseController',{$scope:$scope});//继承

    $scope.search = function(page,rows){
        brandService.search(page,rows,$scope.searchEntity).success(
            function (response) {
                $scope.list=response.rows;	//显示当前页的数据
                $scope.paginationConf.totalItems=response.total;//更新总记录数
            }
        );
    }

    $scope.dele = function(){
        brandService.dele($scope.selectIds).success(
            function (response) {
                if(response.success){
                    $scope.reloadList();
                }else{
                    alert(response.message);
                }
            }
        );
    }




    //根据主键查询
    $scope.findOne = function(id){
        brandService.findOne(id).success(
            function (response) {
                //由于 新增和修改用同一个编辑窗口 所以此处必须和展现对象一致 也就是 pojo
                $scope.pojo = response;
            }
        );
    }
    //保存
    $scope.save = function(){

        var methodStr = "save";

        if($scope.pojo.id != null){
            methodStr = "update";
        }

        brandService.save(methodStr,$scope.pojo).success(
            function (response) {
                if(response.success){
                    //保存成功 则刷新页面 展现新增数据
                    $scope.reloadList();
                }else{
                    alert(response.message);
                }
            }
        );

    }

    //调用后台品牌列表的方法
    $scope.findAll = function () {
        brandService.findAll().success(
            function (response) {
                $scope.list = response;
            }
        );
    }

    //重新加载列表 数据


    //分页
    $scope.findPage=function(page,rows){
        brandService.findPage(page,rows).success(
            function(response){
                $scope.list=response.rows;	//显示当前页的数据
                $scope.paginationConf.totalItems=response.total;//更新总记录数
            }
        );
    }

});

File: dongyimai-manager-web/src/main/java/com/dongyimai/manager/controller/BrandController.java
package com.dongyimai.manager.controller;

import com.alibaba.dubbo.config.annotation.Reference;
import com.dongyimai.entity.Result;
import com.dongyimai.entity.PageResult;
import com.dongyimai.pojo.TbBrand;
import com.dongyimai.sellergoods.service.BrandService;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;
import java.util.Map;

@RestController
@RequestMapping("/brand")
public class BrandController {
    @Reference
    private BrandService brandService;

    @RequestMapping("/dele")
    public Result dele(Long [] ids){
        try {
            brandService.delete(ids);
            return new Result(true,"删除成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false,"删除失败");
        }
    }

    @RequestMapping("/update")
    public Result update(@RequestBody TbBrand tbBrand){
        try {
            brandService.update(tbBrand);
            return new Result(true,"修改成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false,"修改失败");
        }
    }

    @RequestMapping("/findOne")
    public TbBrand findOne(Long bid){
        return brandService.findOne(bid);
    }


    @RequestMapping("/findAll")
    public List<TbBrand> findAll(){
        return brandService.findAll();
    }

    @RequestMapping("/findPage")
    public PageResult findPage(int page,int rows){
        return brandService.findPages(page, rows);
    }

    @RequestMapping("/save")
    public Result save(@RequestBody TbBrand tbBrand){
        try {
            brandService.save(tbBrand);
            return new Result(true,"添加成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false,"添加失败");
        }
    }

    @RequestMapping("/search")
    public PageResult search(@RequestBody TbBrand tbBrand,int page,int rows){
        return brandService.search(tbBrand,page,rows);
    }

    @RequestMapping("/selectOptionList")
    public List<Map> selectOptionList(){
        return brandService.selectOptionList();
    }
}
